src/environmentd/src/http/sql.rs

```diff
@@ -101,6 +106,147 @@ impl Error {
     }
 }
 
+static PER_REPLICA_LABELS: &[&str] = &["replica_full_name", "instance_id", "replica_id"];
+
+async fn execute_promsql_query(
+    client: &mut AuthedClient,
+    query: &PrometheusSqlQuery<'_>,
+    metrics_registry: &MetricsRegistry,
+    metrics_by_name: &mut BTreeMap<String, GenericGaugeVec<AtomicF64>>,
+    cluster: Option<(&Cluster, &ClusterReplica)>,
+) {
+    assert!(query.per_replica == cluster.is_some());
```

Comment:
```suggestion
    assert_eq!(query.per_replica, cluster.is_some());
```

would give a slightly better error message.